FEXCore/include/FEXCore/Core/Context.h

```diff
@@ -194,6 +194,16 @@ class Context {
   ///< Sets FEX's internal EFLAGS representation to the passed in compacted form.
   FEX_DEFAULT_VISIBILITY virtual void SetFlagsFromCompactedEFLAGS(FEXCore::Core::InternalThreadState* Thread, uint32_t EFLAGS) = 0;
 
+  enum class XMMRecoveryMode {
```

Comment:
Does it make sense to drop this enum, and instead infer the recovery mode inside the helpers by checking if `YMM_High != nullptr`? This would simplify the helper signatures a fair bit and remove a failure mode (passing a mode that doesn't match whether you passed a YMM_High pointer)

Reply:
Indeed, that might make sense

Reply:
I've gone ahead and made this change, IMO it's a lot nicer now, see diff https://rosenzweig.io/diff2.txt ... approving, I'm good with this to go in if you're ok with that diff.